lib/explorer/data_frame.ex

```diff
@@ -1086,20 +1148,23 @@ defmodule Explorer.DataFrame do
   def distinct(df, opts \\ [])
 
   def distinct(df, opts) do
-    opts = Keyword.validate!(opts, columns: [], keep_all?: false)
-
-    column_names = names(df)
+    opts = Keyword.validate!(opts, columns: nil, keep_all?: false)
 
     columns =
       case opts[:columns] do
-        [] -> column_names
-        callback when is_function(callback) -> Enum.filter(column_names, callback)
-        columns -> columns
-      end
+        nil ->
+          names(df)
 
-    Enum.each(columns, fn name ->
-      maybe_raise_column_not_found(column_names, name)
-    end)
+        callback when is_function(callback) ->
+          Enum.filter(names(df), callback)
+
+        [] ->
+          raise ArgumentError,
+                "you must provide at least one column or omit the column option to select all columns"
```

Comment:
I am not a big fan of raising on empty lists. I think it is a semantically valid value. Can we instead return the dataframe as is?